File: specification/network/resource-manager/Microsoft.Network/stable/2021-08-01/examples-java/ServiceCommunityList.java
import com.azure.core.util.Context;

/** Samples for BgpServiceCommunities List. */
public final class Main {
    /*
     * x-ms-original-file: specification/network/resource-manager/Microsoft.Network/stable/2021-08-01/examples/ServiceCommunityList.json
     */
    /**
     * Sample code: ServiceCommunityList.
     *
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void serviceCommunityList(com.azure.resourcemanager.AzureResourceManager azure) {
        azure.networks().manager().serviceClient().getBgpServiceCommunities().list(Context.NONE);
    }
}

File: specification/network/resource-manager/Microsoft.Network/stable/2023-06-01/examples-js/ServiceCommunityList.js
const { NetworkManagementClient } = require("@azure/arm-network");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Gets all the available bgp service communities.
 *
 * @summary Gets all the available bgp service communities.
 * x-ms-original-file: specification/network/resource-manager/Microsoft.Network/stable/2023-06-01/examples/ServiceCommunityList.json
 */
async function serviceCommunityList() {
  const subscriptionId = process.env["NETWORK_SUBSCRIPTION_ID"] || "subid";
  const credential = new DefaultAzureCredential();
  const client = new NetworkManagementClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.bgpServiceCommunities.list()) {
    resArray.push(item);
  }
  console.log(resArray);
}
